PWGHF/Tasks/taskMcEfficiency.cxx

```diff
@@ -0,0 +1,290 @@
+  {
+    candidateLoop<false>(candidates, tracks, tracks); // NOTE third argument has to be provided but is not used as template argument is <false>
+  }
+  PROCESS_SWITCH(HfTaskMcEfficiency, processData, "Process data (no MC information needed)", false);
+
+  void processMc(soa::Join<aod::HfCand2Prong, aod::HfSelD0>& candidates, TracksWithSelectionMC& tracks, aod::McParticles& mcParticles, aod::McCollisionLabels&)
+  {
+    candidateLoop<true>(candidates, tracks, mcParticles);
+
+    auto hCandidates = registry.get<StepTHn>(HIST("hCandidates"));
+    auto hTrackablePtEta = registry.get<StepTHn>(HIST("hTrackablePtEta"));
+
+    // lists for optimization
+    std::vector<bool> tracked(mcParticles.size(), false);
+    std::vector<bool> hasITS(mcParticles.size(), false);
+    std::vector<bool> hasTPC(mcParticles.size(), false);
+    std::vector<bool> selected(mcParticles.size(), false);
+    for (auto& track : tracks) {
+      if (track.mcParticleId() >= 0) {
+        tracked[track.mcParticleId()] = true;
+        if (checkTrack(track)) {
+          selected[track.mcParticleId()] = true;
+        }
+        if (track.hasITS()) {
+          hasITS[track.mcParticleId()] = true;
+        }
+        if (track.hasTPC()) {
+          hasTPC[track.mcParticleId()] = true;
+        }
+      }
+    }
+
+    for (auto pdgCode : (std::vector<int>)pdgCodes) {
+      auto mass = RecoDecay::getMassPDG(pdgCode);
+
+      for (auto& mcParticle : mcParticles) {
+        if (mcParticle.pdgCode() != pdgCode) {
+          continue;
+        }
+        hCandidates->Fill(kHFStepMC, mcParticle.pt(), mass, pdgCode, 1.0, true);
+
+        if (mcParticle.daughtersIds().size() != 2) {
+          LOGP(fatal, "Invalid numbers of daughters for D0(bar) {}: {}", mcParticle.globalIndex(), mcParticle.daughtersIds().size());
+        }
+
+        auto prong0Id = mcParticle.daughtersIds()[0];
+        auto prong1Id = mcParticle.daughtersIds()[1];
+
+        if (prong0Id < 0 || prong1Id < 0) {
+          continue;
+        }
+
+        bool inAcceptance = true;
+        for (auto& daugther : mcParticle.daughters_as<aod::McParticles>()) {
+          if (daugther.pt() < mcAcceptancePt || std::abs(daugther.eta()) > mcAcceptanceEta) {
+            inAcceptance = false;
+          }
+        }
+
+        if (inAcceptance) {
+          hCandidates->Fill(kHFStepAcceptance, mcParticle.pt(), mass, pdgCode, 1.0, true);
+        }
+
+        if (tracked[prong0Id] && tracked[prong1Id]) {
+          hCandidates->Fill(kHFStepTrackable, mcParticle.pt(), mass, pdgCode, 1.0, true);
+          if (inAcceptance) {
+            hCandidates->Fill(kHFStepAcceptanceTrackable, mcParticle.pt(), mass, pdgCode, 1.0, true);
+          } else {
+            LOGP(debug, "Candidate {} not in acceptance but tracked.", mcParticle.globalIndex());
+            for (auto& daugther : mcParticle.daughters_as<aod::McParticles>()) {
```

Comment:
`mcParticle.daughters_as<aod::McParticles>()` is used four times at different places in this loop. It would be better to call it just once and store it.

Reply:
> > Thanks for the review, Vit. Please check - everything which is resolved should be done.
> 
> Thanks for addressing the comments. What do you prefer to do with the remaining [#2086 (comment)](https://github.com/AliceO2Group/O2Physics/pull/2086#discussion_r1124732806) ?

Not overlook it :-D